src/components/Match/index.js

```diff
@@ -0,0 +1,106 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import Competitor from './components/Competitor';
+import { declareWinner, removeMatch } from './../../actions/matches';
+import { MatchWrapper } from './styles';
+
+class Match extends Component {
+
+    constructor(props) {
+        super(props);
+        this.handleDeclareWinner = this.handleDeclareWinner.bind(this);
+        this.handleRemoveMatch = this.handleRemoveMatch.bind(this);
+
+    }
+
+
+
+    /**
+     * user can only declare winner, if he is 1 of the competitors
+     * and only 1 time
+     */
+    handleDeclareWinner(competitor) {
+        declareWinner(this.props.matchId, this.props.match, competitor, this.props.auth)
+    }
+
+    /** 
+     * check if the auth has voted for the competitor
+     * so we can show that on the component
+     * the votes are in the match.winners property
+    */
+    checkAuthVote(competitor) {
+        if (this.props.match.winners && this.props.match.winners[this.props.auth.uid]) {
+            if (this.props.match.winners[this.props.auth.uid] === competitor.uid) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * check if the authUsaer is a competitor in this match
+     */
+    checkAuthIsCompetitor(competitors) {
+        let isCompetitor = false;
+        Object.keys(competitors).forEach((key) => {
+            const competitor = competitors[key];
+            if (competitor.uid === this.props.auth.uid) {
+                isCompetitor = true;
+            }
+        });
+        return isCompetitor;
+    }
+
+    handleRemoveMatch() {
+        removeMatch(this.props.matchId)
+            .then(() => {
+                console.log('match removed');
+            });
+    }
+
+    removeMatchTemplate(winner) {
+        if (!winner) {
+            return (
+                <button className="deletebutton" onClick={this.handleRemoveMatch}>
+                    remove match
+                </button>
+            );
+        }
+    }
+
+    render() {
+        const contestedText = this.props.isContested ? 'this match result is contested' : '';
+        return (
+            <MatchWrapper contested={this.props.match.isContested}>
+                <strong>{contestedText}</strong>
+                {this.removeMatchTemplate(this.props.match.winner)}
+                {
+                    Object.keys(this.props.match.competitors).map((key) => {
```

Comment:
Op meer plekken wordt er object.keys aangeroepen. Is het niet een beter idee om de match.competitors een array te laten zijn met competitor objects ?

Reply:
zeker...firebase returned alleen objects ipv arrays.
kan het wel converten, maar dan moet je t ook weer terug converten naar een object als je data wilt bewaren. :-/
Maar wie weet zie ik iets over het hoofd....ik hoop t